### ISSUE TITLE
Do not use Sprintf to convert types

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#873]
Do not use `fmt.Sprintf` to convert types (e.g. `fmt.Sprintf("%d", aNumber)`, `fmt.Sprintf("%t", aBoolean)`)

Before:

```Go
_ = fmt.Sprintf("%d", i)
_ = fmt.Sprintf("%t", i&1 == 0)
```

After:

```Go
_ = strconv.FormatInt(int64(i), 10)
_ = strconv.FormatBool(i&1 == 0)
```

Inspired by [Bad Go: frivolous Sprintf](https://philpearl.github.io/post/bad_go_sprintf/)

**Comment by quasilyte** [comment]
It looks like it can be expressed with AST-based patterns quite easily:

https://github.com/quasilyte/go-ruleguard/blob/master/analyzer/testdata/src/extra/extra.rules.go

I'll copy a small example here:
```go
m.Match(`fmt.Sprintf("%t", $i&1 == 0)`).
	Report(`use strconv.FormatBool($i&1 == 0)`)
```

We might include an ability to load ruleguard-based rules into go-critic later when it's more stable and is actually released, but for now, you can use `go-ruleguard` on its own. :)

If anyone submits a code that implements this check into go-critic (without ruleguard), I'm still willing to merge that PR.

**Comment by Antonboom** [comment]
https://github.com/golangci/golangci-lint/pull/3714
